packages/core/src/html/UlNodeProcessor.ts

```diff
@@ -0,0 +1,189 @@
+import cheerio from 'cheerio';
+import { MbNode, NodeOrText } from '../utils/node';
+
+const { processIconString } = require('../lib/markdown-it/plugins/markdown-it-icons');
+const emojiDictionary = require('../lib/markdown-it/patches/markdown-it-emoji-fixed');
+
+interface EmojiData {
+  [key: string]: string;
+}
+
+const emojiData = emojiDictionary as unknown as EmojiData;
+
+const ICON_ATTRIBUTES = ['icon', 'i-width', 'i-height', 'i-size', 'i-class'];
+
+interface IconAttributes {
+  icon?: string;
+  className?: string;
+  size?: string;
+  width?: string;
+  height?: string;
+}
+
+type IconAttributeDetail = {
+  isFirst: boolean;
+  iconAttr: IconAttributes | null;
+  level: number;
+};
+
+type IconAttributeWrapper = {
+  key: number;
+  value: IconAttributeDetail;
+};
+
+function classifyIcon(icon: string) {
+  const isEmoji = Object.prototype.hasOwnProperty.call(emojiData, icon);
+
+  return {
+    isEmoji,
+    unicodeEmoji: isEmoji
+      ? emojiData[icon]
+      : undefined,
+  };
+}
+
+function createIconSpan(iconAttrs: IconAttributes): cheerio.Cheerio {
+  const {
+    isEmoji,
+    unicodeEmoji,
+  } = classifyIcon(iconAttrs.icon!);
+
+  let spanContent;
+  if (isEmoji) {
+    spanContent = `<span aria-hidden="true">${unicodeEmoji}</span>`;
+  } else {
+    spanContent = processIconString(iconAttrs.icon);
+  }
+
+  let spanNode;
+  if (spanContent === null) {
+    const img = cheerio(`<img src='${iconAttrs.icon}' alt='Icon'>`)
+      .css({ width: iconAttrs.width, height: iconAttrs.height, display: 'inline-block' })
+      .addClass(iconAttrs.className || '');
+    img.append('\u200B');
+
+    spanContent = cheerio('<span></span>').append(img).css({
+      'padding-bottom': '0.3em',
+      'padding-top': '0.3em',
+    });
+    spanNode = cheerio(spanContent).css({ 'font-size': 'unset' });
+  } else {
+    spanNode = cheerio(spanContent).css({ 'font-size': 'unset' });
+    spanNode = spanNode.css({ 'font-size': iconAttrs.size }).addClass(iconAttrs.className || '');
+  }
+  // Add invisible character to avoid the element from being empty
+  spanNode.append('\u200B');
+  return spanNode.css({
+    'line-height': 'unset',
+    'margin-inline-end': '0.3em',
```

Comment:
looks a little tight personally for the images in the docs, would this work well? 

```suggestion
    'margin-inline-end': '0.35em',
```

we can defer this to the future if not (maybe an `i-spacing` setting or similar)

Reply:
Added i-spacing

Reply:
> i-spacing

I should have clarified this was a random, not well thought out suggestion. 

E.g. an alternative is scaling the spacing automatically according to width of the icon, up to a certain maximum. This is a less flexible but "things should just work" approach.

I'd prefer to proceed without and revert this for now to get this merged, then reapproach this discussion later once this has been used for a while. (you can open an issue referencing your commit and this comment)

But if you'd prefer to handle it now, let's get @damithc's input as well.

Reply:
> I'd prefer to proceed without and revert this for now to get this merged, then reapproach this discussion later once this has been used for a while. (you can open an issue referencing your commit and this comment)

Yup, let's do that.